ConvertTsv.py

```diff
@@ -2,96 +2,119 @@
-            f.write(text)
 
-    path = './data/aclImdb/test/neg/'
-
-    for fname in glob.glob(os.path.join(path, '*.txt')):
-        with io.open(fname, 'r', encoding="utf-8") as ff:
-            text = ff.readline()
-            text = text.replace('\t', " ")
-            text = text + '\t' + '0' + '\t' + '\n'
-            f.write(text)
+def get_IMDb_DataLoaders_and_TEXT(max_length=256, batch_size=24, debug_log=False):
+    filenames = glob.glob(base + '/data-japanese/text/*/*')
+    # filenames = [base + '/data-japanese/text/kaden-channel/kaden-channel-5774093.txt',
+    #              base + '/data-japanese/text/kaden-channel/kaden-channel-5774562.txt']
+    # print(filenames,len(filenames))
+    keys = []
+    texts = []
+    labels = []
+    for filename in filenames:
+        if re.search('LICENSE', filename):
+            continue
+
+        with open(filename, 'r', encoding='utf-8') as f:
+            filepath, basename = os.path.split(filename)
+
+            keys.append(os.path.splitext(basename)[0])
+            texts.append(''.join(f.readlines()[2:]))
+            labels.append(os.path.split(filepath)[-1])
+
+    news = pd.DataFrame({'key': keys, 'label': labels, 'text': texts})
+
+    # natoo-pyではコンストラクタに渡す
+
+    mecab_sym = MeCab("-Ochasen")
+    print(mecab_sym.parse('こんにちはかなちゃん。'))
+    a = 'こんにちはかなちゃん。'
+    print(mecab_sym.parse(a).split("\n"))
+    symbol_list = []
+
+    def pick_sym(text):
+        node = mecab_sym.parse(text)
+        node_list = node.split("\n")
+        for node in node_list:
+            if '記号' in node:
+                symbol_list.append(node[0])
+
+    news['text'] = news['text'].apply(mojimoji.han_to_zen)
+
+    symbol_list = []
+    for i in range(len(news)):
+        symbol_list.append(pick_sym(news['text'].iloc[i]))
+
+    symbol_list = list(set(symbol_list))
+    print(symbol_list)
+
+    def del_sym(df):
+        # print(df['text'])
+        df['text'] = df['text'].replace('０', '')
+        df['text'] = df['text'].replace('１', '')
+        df['text'] = df['text'].replace('２', '')
+        df['text'] = df['text'].replace('３', '')
+        df['text'] = df['text'].replace('４', '')
+        df['text'] = df['text'].replace('５', '')
+        df['text'] = df['text'].replace('６', '')
+        df['text'] = df['text'].replace('７', '')
+        df['text'] = df['text'].replace('８', '')
+        df['text'] = df['text'].replace('９', '')
+        df['text'] = df['text'].replace('\d+年', '')
+        df['text'] = df['text'].replace('\d+月', '')
+        df['text'] = df['text'].replace('\d+日', '')
+        df['text'] = df['text'].replace('\d', '')
+        df['text'] = df['text'].replace('\n', '')
+
+        for i in range(len(symbol_list)):
+            if not symbol_list[i]:
+                continue
+            df['text'] = df['text'].replace(symbol_list[i], '')
+        return df
 
-    f.close()
+    # いらないものを取り去り、tsvに格納する作業
+    for DataFrameTextIndex in range(len(news)):
```

Comment:
pandasのDataFrameでは基本的にfor文での処理が遅くなるので、`DataFrame.apply`のようなメソッドを使うことをおすすめします。今回は `text` カラムが対象なので、次のように書けそうです。
```suggestion
    news["text"] = news["text"].apply(del_sym)
```

Reply:
この書き方とても参考になります！